src/saltext/vmware/states/vmc_nat_rules.py

```diff
@@ -0,0 +1,486 @@
+            )
+
+            if "error" in updated_nat_rule:
+                return vmc_state._create_state_response(
+                    name=name, comment=updated_nat_rule["error"], result=False
+                )
+
+            get_nat_rule_after_update = __salt__["vmc_nat_rules.get_by_id"](
+                hostname=hostname,
+                refresh_key=refresh_key,
+                authorization_host=authorization_host,
+                org_id=org_id,
+                sddc_id=sddc_id,
+                tier1=tier1,
+                nat=nat,
+                nat_rule=nat_rule,
+                verify_ssl=verify_ssl,
+                cert=cert,
+            )
+
+            if "error" in get_nat_rule_after_update:
+                return vmc_state._create_state_response(
+                    name=name, comment=get_nat_rule_after_update["error"], result=False
+                )
+
+            return vmc_state._create_state_response(
+                name=name,
+                comment="Updated nat rule {}".format(nat_rule),
+                old_state=existing_nat_rule,
+                new_state=get_nat_rule_after_update,
+                result=True,
+            )
+        else:
+            log.info("All fields are same as existing nat rule %s", nat_rule)
+            return vmc_state._create_state_response(
+                name=name, comment="Nat rule exists already, no action to perform", result=True
+            )
+    else:
+        log.info("No nat rule found with Id %s", nat_rule)
+        created_nat_rule = __salt__["vmc_nat_rules.create"](
+            hostname=hostname,
+            refresh_key=refresh_key,
+            authorization_host=authorization_host,
+            org_id=org_id,
+            sddc_id=sddc_id,
+            tier1=tier1,
+            nat=nat,
+            nat_rule=nat_rule,
+            verify_ssl=verify_ssl,
+            cert=cert,
+            action=action,
+            destination_network=destination_network,
+            source_network=source_network,
+            translated_network=translated_network,
+            translated_ports=translated_ports,
+            scope=scope,
+            service=service,
+            enabled=enabled,
+            firewall_match=firewall_match,
+            logging=logging,
+            description=description,
+            tags=tags,
+            sequence_number=sequence_number,
+        )
+
+        if "error" in created_nat_rule:
+            return vmc_state._create_state_response(
+                name=name, comment=created_nat_rule["error"], result=False
+            )
+
+        return vmc_state._create_state_response(
+            name=name,
+            comment="Created nat rule {}".format(nat_rule),
+            new_state=created_nat_rule,
+            result=True,
+        )
+
+
+def absent(
+    name,
+    hostname,
+    refresh_key,
+    authorization_host,
+    org_id,
+    sddc_id,
+    tier1,
+    nat,
+    nat_rule,
```

Comment:
**suggestion**: same comment applies Re: name vs nat_rule. For states the `name` is the primary identifier of the thing that the state is controlling. In this case, `name` should be "Id of the security_rule to be deleted from SDDC". Also, is `security_rule` defined in the SDDC docs?

Reply:
As discussed, changes will be made as part of the issue: https://github.com/saltstack/salt-ext-modules-vmware/issues/120 for all states